File: web-unblocker/Nodejs/proxy.js
const settings = require('./settings');

module.exports = {
  createProxyByUrl: (url) => {
    let country
    let parsedUrl = url

    urlParts = url.split(';')
    if (urlParts.length === 2) {
      parsedUrl = urlParts[0]
      country = urlParts[1]
    }

    return [
      parsedUrl,
      country,
      `http://${settings.Username}:${settings.Password}@${settings.ProxyAddress}`
    ]
  }
}

File: web-unblocker/GoLang/proxy.go
package main

import (
	"fmt"
	"strings"
)

func createProxyByUrl(url string) (string, *string, string) {
	var country *string
	parsedUrl := url

	urlParts := strings.Split(url, ";")
	if len(urlParts) == 2 {
		parsedUrl = urlParts[0]
		country = &urlParts[1]
	}

	return parsedUrl, country, fmt.Sprintf(
		"http://%s:%s@%s",
		Username,
		Password,
		ProxyAddress,
	)
}
